nautilus/src/nautilus/tracing/TracingUtil.cpp

```diff
@@ -79,9 +57,10 @@ TypedValueRef& traceBinaryOp(Op operation, Type resultType, const TypedValueRef&
 
 TypedValueRef& traceCall(void* fptn, Type resultType, const std::vector<tracing::TypedValueRef>& arguments,
                          const FunctionAttributes fnAttrs) {
-	auto mangledName = getMangledName(fptn);
-	auto functionName = getFunctionName(mangledName);
-	return TraceContext::get()->traceCall(functionName, mangledName, fptn, resultType, arguments, fnAttrs);
+	auto traceContext = TraceContext::get();
+	auto mangledName = traceContext->getMangledName(fptn);
+	auto functionName = traceContext->getFunctionName(mangledName);
+	return traceContext->traceCall(functionName, mangledName, fptn, resultType, arguments, fnAttrs);
```

Comment:
lets move 
auto mangledName = traceContext->getMangledName(fptn);
	auto functionName = traceContext->getFunctionName(mangledName);
to traceContext->traceCall()